src/Algebra/Apartness/Properties/HeytingField.agda

```diff
@@ -0,0 +1,85 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of Heyting Fields
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Apartness.Bundles using (HeytingField)
+
+module Algebra.Apartness.Properties.HeytingField
+  {c ℓ₁ ℓ₂} (HF : HeytingField c ℓ₁ ℓ₂) where
+
+open import Function.Base using (_∘_)
+open import Data.Product.Base using (_,_; proj₁; proj₂)
+open import Algebra.Bundles using (CommutativeRing)
+
+open HeytingField HF
+open CommutativeRing commutativeRing using (ring; *-commutativeMonoid)
+
+open import Algebra.Definitions _≈_
+  using (Invertible; LeftInvertible; RightInvertible)
+open import Algebra.Properties.CommutativeMonoid *-commutativeMonoid
+  using (invertibleˡ⇒invertible; invertibleʳ⇒invertible)
+open import Algebra.Properties.Ring ring
+  using (x-0#≈x; -‿distribˡ-*; -‿distribʳ-*; -‿anti-homo-+; -‿involutive)
+open import Relation.Binary.Definitions using (Symmetric)
+import Relation.Binary.Reasoning.Setoid as ≈-Reasoning
+
+private
+  variable
+    x y z : Carrier
+
+
+invertibleˡ⇒# : LeftInvertible 1# _*_ (x - y) → x # y
+invertibleˡ⇒# = invertible⇒# ∘ invertibleˡ⇒invertible
+
+invertibleʳ⇒# : RightInvertible 1# _*_ (x - y) → x # y
+invertibleʳ⇒# = invertible⇒# ∘ invertibleʳ⇒invertible
+
+1#0 : 1# # 0#
+1#0 = invertibleˡ⇒# (1# , 1*[x-0]≈x)
+  where
+  1*[x-0]≈x : 1# * (x - 0#) ≈ x
+  1*[x-0]≈x {x} = trans (*-identityˡ (x - 0#)) (x-0#≈x x)
+
+x#0y#0→xy#0 : x # 0# → y # 0# → x * y # 0#
+x#0y#0→xy#0 {x} {y} x#0 y#0 = helper (#⇒invertible x#0) (#⇒invertible y#0)
```

Comment:
@JacquesCarette 's wanting to refactor the proofs. 
It occurs to me that some (all?) of these arguments could be refactored by going via `Algebra.Properties.CancellativeCommutativeSemiring`, after establishing the relevant structure/bundle?

Reply:
Could be. I was thinking of analogs to the `elim` and `cancel` reasoning combinators of `agda-categories`. While at it, I would have ported push/pull (but renamed them) too. Many of these already work on `Monoid`.

Reply:
Hoping to do that via proxy (if in the next month) or perhaps myself after that.